ferry.go

```diff
@@ -107,7 +174,14 @@ func (f *Ferry) Initialize() (err error) {
 
 	f.logger.Infof("hello world from %s", VersionString)
 
-	// Connect to the database
+	// Suppress siddontang/go-mysql logging as we already log the equivalents.
+	// It also by defaults logs to stdout, which is different from Ghostferry
+	// logging, which all goes to stderr. stdout in Ghostferry is reserved for
+	// dumping states due to an abort.
```

Comment:
This seems a bit fragile. What if one of the other dependencies that we use logs to stdout? Or maybe a future one that we want to use does that and doesn't allow the user to disable it? What do we do then?

Reply:
My current thinking on this is that stdout is the most flexible method to dump the state with the least amount of code within Ghostferry itself. 

If in the future we encounter issues with stdout, we can definitely make Ghostferry write to a file/pipe/whatever. We should deal with it when it comes up, especially since we don't quite have a vision on how to consume this state dump just yet.